Library/Homebrew/formula_installer.rb

```diff
@@ -942,14 +942,41 @@ def post_install
     @show_summary_heading = true
   end
 
-  def pour
+  def fetch_dependencies
+    deps = compute_dependencies
+
+    return if deps.empty? || ignore_deps?
+
+    deps.each do |dep|
+      fetch_dependency(dep.to_formula)
+    end
+  end
+
+  def fetch_dependency(dep)
+    dep.resources.each do |resource|
+      resource.fetch(verify_download_integrity: false)
+    end
+  end
+
+  def fetch
+    fetch_dependencies
+
+    if pour_bottle?
+      downloader.fetch
+    else
+      formula.downloader.fetch
+    end
```

Comment:
```suggestion
    if pour_bottle?
      downloader.fetch
    else
      formula.downloader.fetch
    end
```

Move this logic to `downloader` so it returns `formula.downloader.fetch` if `pour_bottle?` and `formula.local_bottle_path` aren't set.